### PR TITLE
optim get schema and get class name in shards into 1 rdd operation

### PR DESCRIPTION
**Pull request by dding3**
## Description

optimize  `_get_class_name` and `_get_schema` into one rdd.map operation.

### 1. Why the change?

In current implementation, `_get_class_name` and `_get_schema` are get with 2 rdd operations which will bring extra overhead, put them into one to save cost.

### 2. User API changes

N/A

### 3. How to test?
- [x] Unit test
- [x] Application test

### CODE REVIEW

File: python/orca/src/bigdl/orca/data/shard.py [hunk 1]
```diff
@@ -454,20 +454,29 @@ def _get_schema(self):
         if 'schema' in self.type:
             return self.type['schema']
         else:
-            if self._get_class_name() == 'pandas.core.frame.DataFrame':
-                import pandas as pd
-                columns, dtypes = self.rdd.map(lambda x: (x.columns, x.dtypes)).first()
-                self.type['schema'] = {'columns': columns, 'dtypes': dtypes}
-                return self.type['schema']
-            return None
+            class_name, schema = self._get_schema_class_name()
```
Comment (jason-dai) [inline python/orca/src/bigdl/orca/data/shard.py:457]: only call `_get_schema_class_name` if `class_name` does not exist, or is pandas dataframe

**Comment by dding3** [comment]
most jekins passed. http://10.112.231.51:18888/view/ZOO-PR/job/BigDL-Orca-PR-Validation/1011/.  http://10.112.231.51:18888/job/Orca-PRVN-Others/1015/console

The failed one is not related to this pr.

File: python/orca/src/bigdl/orca/data/shard.py [hunk 2]
```diff
@@ -598,21 +598,32 @@ def utility_func(x, func, *args, **kwargs):
...
-                self.type['schema'] = {'columns': columns, 'dtypes': dtypes}
-                return self.type['schema']
-            return None
+
+        if 'class_name' not in self.type\
+                or self.type['class_name'] == 'pandas.core.frame.DataFrame':
+            class_name, schema = self._get_schema_class_name()
+            self.type['class_name'] = class_name
+            self.type['schema'] = schema
+            return self.type['schema']
```
Comment (jason-dai) [inline python/orca/src/bigdl/orca/data/shard.py:607]: Need to return None otherwise

Comment (dding3) [reply python/orca/src/bigdl/orca/data/shard.py:607]: Fixed in https://github.com/intel-analytics/BigDL/pull/5558

**Pull request merged**